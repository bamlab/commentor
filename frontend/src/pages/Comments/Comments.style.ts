import styled from 'styled-components';
import { colorUsage } from 'stylesheet';

const StyledComments = styled.div`
  display: flex;
  flex-direction: column;
`;

export const HeaderText = styled.p`
  font-weight: bold;
`;

export const HeaderCell = styled.div`
  background-color: ${colorUsage.headerBackground};
`;

export const FloatingButtonContainer = styled.div`
  position: fixed;
  right: 2%;
  bottom: 2%;
`;

export default StyledComments;
