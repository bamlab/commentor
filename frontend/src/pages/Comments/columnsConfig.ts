import { ColumnType } from 'components/GenericTable/GenericTable.type';

export const columnsConfig: ColumnType[] = [
  {
    index: 0,
    key: 'id',
    name: 'id',
    columnWidth: 75,
  },
  {
    index: 1,
    key: 'body',
    name: 'body',
    columnWidth: 300,
  },
  {
    index: 2,
    key: 'filePath',
    name: 'filePath',
    columnWidth: 300,
  },
  {
    index: 3,
    key: 'url',
    name: 'url',
    columnWidth: 300,
  },
  {
    index: 4,
    key: 'commentor',
    name: 'commentor',
    columnWidth: 200,
  },
  {
    index: 5,
    key: 'pullRequestUrl',
    name: 'pullRequestUrl',
    columnWidth: 200,
  },
  {
    index: 6,
    key: 'repositoryId',
    name: 'repositoryId',
    columnWidth: 200,
  },
  {
    index: 6,
    key: 'creationDate',
    name: 'creationDate',
    columnWidth: 200,
  },
];

export const fixedColumnCount = 2; // this is the reference
