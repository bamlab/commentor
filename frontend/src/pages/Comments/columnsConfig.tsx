import React from 'react';
import { ColumnType } from 'components/GenericTable/GenericTable.type';
import { Cell, StandardTextContainer } from 'components/GenericTable/GenericTable.style';
import LinkCellIcon from './components/LinkCellIcon';

export type CommentTableOptionsType = {};

const defaultCommentCellRenderer = (
  key: string,
  value: string,
  objectId: number | 'id',
  style: Object,
): JSX.Element => {
  return (
    <Cell key={key} style={style}>
      <StandardTextContainer>{value}</StandardTextContainer>
    </Cell>
  );
};

export const columnsConfig: ColumnType<CommentTableOptionsType>[] = [
  {
    index: 0,
    key: 'body',
    name: 'Comment',
    columnWidth: 450,
    renderer: (key: string, value: any, objectId: number | 'id', style: Object): JSX.Element =>
      defaultCommentCellRenderer(key, value, objectId, style),
  },
  {
    index: 1,
    key: 'filePath',
    name: 'File',
    columnWidth: 300,
    renderer: (key: string, value: any, objectId: number | 'id', style: Object): JSX.Element =>
      defaultCommentCellRenderer(key, value, objectId, style),
  },
  {
    index: 2,
    key: 'commentor',
    name: 'Commentor',
    columnWidth: 200,
    renderer: (key: string, value: any, objectId: number | 'id', style: Object): JSX.Element =>
      defaultCommentCellRenderer(key, value, objectId, style),
  },
  {
    index: 3,
    key: 'creationDate',
    name: 'Creation Date',
    columnWidth: 200,
    renderer: (key: string, value: any, objectId: number | 'id', style: Object): JSX.Element =>
      defaultCommentCellRenderer(key, value, objectId, style),
  },
  {
    index: 4,
    key: 'url',
    name: 'url',
    columnWidth: 150,
    renderer: (key: string, value: any, objectId: number | 'id', style: Object): JSX.Element =>
      LinkToComment(key, value, style),
  },
  {
    index: 4,
    key: 'pullRequestUrl',
    name: 'pullRequestUrl',
    columnWidth: 150,
    renderer: (key: string, value: string, objectId: number | 'id', style: Object): JSX.Element =>
      LinkToPullRequest(key, value, style),
  },
];

const LinkToPullRequest = (
  key: string,
  linkURL: string | 'pullRequestUrl',
  style: Object,
): JSX.Element => {
  return (
    <Cell key={key} style={style}>
      {linkURL !== 'pullRequestUrl' ? (
        <LinkCellIcon onIconClick={() => window.open(linkURL, '_blank')} />
      ) : (
        'Go to PR'
      )}
    </Cell>
  );
};

const LinkToComment = (key: string, linkURL: string | 'url', style: Object): JSX.Element => {
  return (
    <Cell key={key} style={style}>
      {linkURL !== 'url' ? (
        <LinkCellIcon onIconClick={() => window.open(linkURL, '_blank')} />
      ) : (
        'Go to Comment'
      )}
    </Cell>
  );
};
export const lineHeight = 700;
export const fixedColumnCount = 2; // this is the reference
