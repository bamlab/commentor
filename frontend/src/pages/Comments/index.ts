export { default } from './Comments.wrap';
