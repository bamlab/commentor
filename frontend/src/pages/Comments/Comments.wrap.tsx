import { connect } from 'react-redux';
import Comments from './Comments';
import { getComments, isCommentLoading } from 'redux/Comment/comment.selectors';
import { loadComments } from 'redux/Comment/comment.actions';
import { Dispatch } from 'redux';
import { RootState } from 'redux/types';
import { getSelectedRepositoryIds } from 'redux/Filters';

const mapStateToProps = (state: RootState) => ({
  comments: getComments(state),
  isCommentLoading: isCommentLoading(state),
  repositoryIds: getSelectedRepositoryIds(state),
});

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    loadComments: (filters: { repositoryIds: number[] }) =>
      dispatch(
        loadComments.request({
          ...filters,
        }),
      ),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Comments);
