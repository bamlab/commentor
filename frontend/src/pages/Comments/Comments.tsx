import React, { useEffect } from 'react';
import { FormattedMessage } from 'react-intl';
import Loader from 'components/Loader';
import StyledComments, { LoaderContainer, ButtonContainer } from './Comments.style';
import { CommentType } from 'redux/Comment';
import { GenericTable } from 'components/GenericTable/GenericTable';
import Button from 'components/Button';
import { fixedColumnCount, columnsConfig } from './columnsConfig';
interface IProps {
  comments: CommentType[];
  loadComments: () => void;
  isCommentLoading: boolean;
}

const Comments = React.memo<IProps>(props => {
  useEffect(() => {
    props.loadComments();
    // eslint-disable-next-line
  }, []);

  return (
    <StyledComments>
      {props.isCommentLoading && (
        <LoaderContainer>
          <Loader />
        </LoaderContainer>
      )}
      <FormattedMessage id="comments.tmp" />
      <GenericTable
        values={props.comments}
        fixedColumnCount={fixedColumnCount}
        columnsConfig={columnsConfig}
      />
      <ButtonContainer>
        <Button disabled={props.isCommentLoading} onClick={props.loadComments}>
          Refresh Comments
        </Button>
      </ButtonContainer>
    </StyledComments>
  );
});

export default Comments;
