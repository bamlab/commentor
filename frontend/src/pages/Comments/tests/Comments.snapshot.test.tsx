import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Comments from '../Comments';

/*
Snapshot tests allow you to easily lock the comportment of a component.
Given props, it renders the component and compares it to the saved snapshot.
*/
describe('[Snapshot] <Comments />', () => {
  it('should render a button with a label', () => {
    const props = {
      intl: {
        formatMessage: jest.fn(),
      },
    };
    const wrapper = shallow(<Comments {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
