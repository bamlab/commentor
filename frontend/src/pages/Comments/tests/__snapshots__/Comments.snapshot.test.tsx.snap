// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Snapshot] <Comments /> should comments page 1`] = `
<Commentsstyle__StyledComments>
  <Commentsstyle__LoaderContainer>
    <Loader />
  </Commentsstyle__LoaderContainer>
  <FormattedMessage
    id="comments.tmp"
    values={Object {}}
  />
  <GenericTable
    columnsConfig={
      Array [
        Object {
          "columnWidth": 75,
          "index": 0,
          "key": "id",
          "name": "id",
        },
        Object {
          "columnWidth": 300,
          "index": 1,
          "key": "body",
          "name": "body",
        },
        Object {
          "columnWidth": 300,
          "index": 2,
          "key": "filePath",
          "name": "filePath",
        },
        Object {
          "columnWidth": 300,
          "index": 3,
          "key": "url",
          "name": "url",
        },
        Object {
          "columnWidth": 200,
          "index": 4,
          "key": "commentor",
          "name": "commentor",
        },
        Object {
          "columnWidth": 200,
          "index": 5,
          "key": "pullRequestUrl",
          "name": "pullRequestUrl",
        },
        Object {
          "columnWidth": 200,
          "index": 6,
          "key": "repositoryId",
          "name": "repositoryId",
        },
        Object {
          "columnWidth": 200,
          "index": 6,
          "key": "creationDate",
          "name": "creationDate",
        },
      ]
    }
    fixedColumnCount={2}
    values={Array []}
  />
  <Commentsstyle__ButtonContainer>
    <Button
      disabled={true}
      onClick={[MockFunction]}
    >
      Refresh Comments
    </Button>
  </Commentsstyle__ButtonContainer>
</Commentsstyle__StyledComments>
`;
